pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14290899?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14290899?buildTab=artifacts#/) on master @ [559994e230ac3776dd1188e66678cce9205ab345](https://github.com/cockroachdb/cockroach/commits/559994e230ac3776dd1188e66678cce9205ab345):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/da69a73e19420a8109697ab7be6f5bb3/logTestSQLStatsCompactor3242106392
    test_log_scope.go:81: use -show-logs to present logs inline
    compaction_test.go:284: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/da69a73e19420a8109697ab7be6f5bb3/logTestSQLStatsCompactor3242106392
--- FAIL: TestSQLStatsCompactor (38.67s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conditional_wrap.go:190: 
        pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:215: (TestSQLStatsCompactor.func1)
        	NOTICE: .AppStopper() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().AppStopper() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    compaction_test.go:251: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:251
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 33
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1
        	Messages:   	expected 32 number of wide scans issued, but 33 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1 (8.12s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36468

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://tanzanite.cluster.engflow.com/invocations/default/03a7b478-b0b1-4f62-a929-83c88632724f?testReportRun=7&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NxbHN0YXRzL3BlcnNpc3RlZHNxbHN0YXRzOnBlcnNpc3RlZHNxbHN0YXRzX3Rlc3Q=) on master @ [559994e230ac3776dd1188e66678cce9205ab345](https://github.com/cockroachdb/cockroach/commits/559994e230ac3776dd1188e66678cce9205ab345):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestSQLStatsCompactor2296797446
    test_log_scope.go:81: use -show-logs to present logs inline
    compaction_test.go:284: -- test log scope end --
test logs left over in: outputs.zip/logTestSQLStatsCompactor2296797446
--- FAIL: TestSQLStatsCompactor (41.55s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conditional_wrap.go:190: 
        pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:215: (TestSQLStatsCompactor.func1)
        	NOTICE: .AppStopper() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().AppStopper() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    compaction_test.go:251: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:251
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 35
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 35 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (6.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
@@ -35,7 +35,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats/persistedsqlstats"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -74,8 +73,6 @@ func TestSQLStatsCompactorNilTestingKnobCheck(t *testing.T) {
 func TestSQLStatsCompactor(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-
-	skip.WithIssue(t, 120092)
 
 	ctx := context.Background()
 
@@ -216,7 +213,7 @@ func TestSQLStatsCompactor(t *testing.T) {
 			}
 
 			generateFingerprints(t, sqlConn, tc.stmtCount)
-			serverSQLStats.Flush(ctx, srv.AppStopper())
+			serverSQLStats.Flush(ctx, server.AppStopper())
 
 			sqlStatsKnobs := sqlstats.CreateTestingKnobs()
 			sqlStatsKnobs.OnCleanupStartForShard = cleanupInterceptor.intercept
@@ -252,12 +249,11 @@ func TestSQLStatsCompactor(t *testing.T) {
 
 			actualNumberOfWideScans := kvInterceptor.getTotalWideScans()
 
-			require.Equal(t,
-				expectedNumberOfWideScans,
-				actualNumberOfWideScans,
-				"expected %d number of wide scans issued, but %d number of "+
-					"wide scan issued", expectedNumberOfWideScans, actualNumberOfWideScans,
-			)
+			if expectedNumberOfWideScans != actualNumberOfWideScans {
+				t.Fatalf("expected %d number of wide scans issued, but %d number of "+
+					"wide scan issued\ndetails: %v", expectedNumberOfWideScans,
+					actualNumberOfWideScans, kvInterceptor.wideScanDetails)
+			}
 
 			actualStmtFingerprints, actualTxnFingerprints :=
 				getTopSortedFingerprints(t, sqlConn, 0 /* limit */)
@@ -395,11 +391,15 @@ func generateFingerprints(t *testing.T, sqlConn *sqlutils.SQLRunner, distinctFin
 	}
 }
 
-var kvReqWideScanStartKeyPattern = regexp.MustCompile(`(/Tenant/\d+)?/Table/\d+/[0-9]{1,2}/[0-9]$`)
+// Tables 42 and 43 and are the IDs of the statement and transactions
+// tables respectively. See `StatementStatisticsTableID` and
+// `TransactionStatisticsTableID`.
+var kvReqWideScanStartKeyPattern = regexp.MustCompile(`(/Tenant/\d+)?/Table/((42)|(43))/[0-9]{1,2}/[0-9]$`)
 
 type kvScanInterceptor struct {
-	totalWideScan int64
-	enabled       int32
+	totalWideScan   int64
+	enabled         int32
+	wideScanDetails []string
 
 	mu struct {
 		syncutil.Mutex
@@ -412,6 +412,7 @@ type kvScanInterceptor struct {
 
 func (k *kvScanInterceptor) reset() {
 	atomic.StoreInt64(&k.totalWideScan, 0)
+	k.wideScanDetails = []string{}
 }
 
 func (k *kvScanInterceptor) getTotalWideScans() int64 {
@@ -443,6 +444,7 @@ func (k *kvScanInterceptor) intercept(ctx context.Context, ba *kvpb.BatchRequest
 			keyMatchedWideScan := kvReqWideScanStartKeyPattern.MatchString(prettyKey)
 
 			if keyMatchedWideScan {
+				k.wideScanDetails = append(k.wideScanDetails, fmt.Sprintf("wide scan in %v", ba))
 				atomic.AddInt64(&k.totalWideScan, 1)
 			}
 		}

